File: backend/src/main/java/org/example/backend/recipe/model/RecipeDetailModelDto.java
package org.example.backend.recipe.model;

import java.util.List;

public record RecipeDetailModelDto(
        String title,
        List<IngredientGroupModel> ingredients,
        List<StepGroupModel> steps,
        NutritionValuesModel nutritionValues,
        int preparationTime,
        int cookingTime,
        int defaultPortions,
        List<String> tags,
        String image
) {
}

File: frontend/src/models/recipe/RecipeDetailModelDto.ts
import type { IngredientGroupModel } from "./IngredientGroupModel.ts";
import type { StepGroupModel } from "./StepGroupModel.ts";
import type { NutritionValuesModel } from "./NutritionValuesModel.ts";

export type RecipeDetailModelDto = {
  title: string;
  ingredients: IngredientGroupModel[];
  steps: StepGroupModel[];
  nutritionValues: NutritionValuesModel;
  preparationTime: number;
  cookingTime: number;
  defaultPortions: number;
  tags: string[];
  image: string;
};
